tool/teleport/common/debug.go

```diff
@@ -0,0 +1,289 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package common
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/lib/config"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/srv/debug"
+	logutils "github.com/gravitational/teleport/lib/utils/log"
+)
+
+// debugServiceClient debug service client.
+type debugServiceClient struct {
+	clt        *http.Client
+	dataDir    string
+	socketPath string
+}
+
+// newDebugServiceClient generates a new debug service client.
+func newDebugServiceClient(configPath string) (*debugServiceClient, error) {
+	cfg, err := config.ReadConfigFile(configPath)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	// ReadConfigFile returns nil configuration if the file doesn't exists.
+	// In that case, fallback to default data dir path.
+	dataDir := defaults.DataDir
+	if cfg != nil {
+		dataDir = cfg.DataDir
+	}
+
+	socketPath := filepath.Join(dataDir, debug.ServiceSocketName)
+	return &debugServiceClient{
```

Comment:
That wouldn't make much difference since we're testing the "CLI functions". If the CLI flow requires the config file, we'll need to provide it anyway.